File: kotlin-electron/karakum/plugins/convertNodeRequire.js
import ts from "typescript";

export default function (node) {
    if (
        ts.isInterfaceDeclaration(node)
        && (
            node.name.text === "NodeRequire"
            || node.name.text === "NodeRequireFunction"
        )
    ) {
        return ""
    }
    return null
}

File: kotlin-electron/karakum/src/jsMain/kotlin/electron/karakum/plugins/convertNodeRequire.kt
package electron.karakum.plugins

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import typescript.isInterfaceDeclaration

val convertNodeRequire = createPlugin { node, _, _ ->
    nullable {
        ensure(isInterfaceDeclaration(node))

        ensure(
            node.name.text == "NodeRequire"
                    || node.name.text == "NodeRequireFunction"
        )

        ""
    }
}
